### PR TITLE
fix(operator_monitoring): fix log parsing

### PR DESCRIPTION
**Pull request by dkropachev**
Operator 1.4 changed it's logging fix log monitoring
 to match new style

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] ~~New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- [ ] ~~I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~~
- [x] All new and existing unit tests passed (CI)
- [ ] ~~I have updated the Readme/doc folder accordingly (if needed)~~

### CODE REVIEW

File: sdcm/sct_events/operator.py [hunk 1]
```diff
@@ -28,74 +30,41 @@ class ScyllaOperatorLogEvent(LogEvent):
...
     "REAPPLY", severity=Severity.WARNING,
     regex="please apply your changes to the latest version and try again")
 ScyllaOperatorLogEvent.add_subevent_type(
     "TLS_HANDSHAKE_ERROR", severity=Severity.WARNING,
-    regex="TLS handshake error from .*: EOF")
+    regex="TLS handshake error from .*")
 ScyllaOperatorLogEvent.add_subevent_type(
     "OPERATOR_STARTED_INFO", severity=Severity.NORMAL,
-    regex="Starting the operator...")
+    regex="] operator version ")
```
Comment (vponomaryov) [inline sdcm/sct_events/operator.py:68]: Here is how looks like line where operator really starts:

```
...
I0816 08:45:02.840256       1 scyllacluster/controller.go:205] "Starting controller" controller="ScyllaCluster"
...
```

File: unit_tests/test_sct_events_operator.py [hunk 2]
```diff
@@ -66,10 +66,11 @@ def test_scylla_operator_log_event_operator_started_info(self):
...
+        assert event.timestamp == 1624884782.0
         self.assertEqual(
-            "(ScyllaOperatorLogEvent Severity.NORMAL) period_type=one-time "
-            "event_id=9bb2980a-5940-49a7-8b08-d5c323b46aa9: "
-            "type=OPERATOR_STARTED_INFO regex=Starting the operator... "
-            f"line_number=0 node=N/A\n{log_record} None None: None, None",
+            '(ScyllaOperatorLogEvent Severity.NORMAL) period_type=one-time '
+            'event_id=9bb2980a-5940-49a7-8b08-d5c323b46aa9: '
+            'type=OPERATOR_STARTED_INFO regex=] operator version :'
+            ' operator/operator.go:133] operator version v1.4.0-alpha.0-34-gf13771d',
```
Comment (vponomaryov) [inline unit_tests/test_sct_events_operator.py:74]: Need to update it considering above comment.

**Review by vponomaryov — CHANGES_REQUESTED** [review]
Please see inline comments.
Also, need to update commit message saying that new logging format appeared in operator-1.4 , not "1.5".

File: sdcm/sct_events/operator.py (hunk 1 above)
Comment (dkropachev) [reply sdcm/sct_events/operator.py]: fixed

File: unit_tests/test_sct_events_operator.py (hunk 2 above)
Comment (dkropachev) [reply unit_tests/test_sct_events_operator.py]: fixed

**Review by vponomaryov — CHANGES_REQUESTED** [review]
Commit message concern was not addressed.

**Comment by dkropachev** [comment]
> Commit message concern was not addressed.

Commit message fixed

**Review by vponomaryov — DISMISSED** [review]

File: sdcm/sct_events/operator.py [hunk 3]
```diff
@@ -28,74 +28,49 @@ class ScyllaOperatorLogEvent(LogEvent):
...
-                continue
-            if attr_name == 'timestamp':
-                log_en_data = datetime.strptime(log_en_data, "%Y-%m-%dT%H:%M:%S.%fZ").timestamp()
-            event_data[attr_name] = log_en_data
+        self.source = source
+        self.line_number = None
+
+    # pylint: disable=too-many-locals
+    def add_info(self: T_log_event, node, line: str, line_number: int) -> T_log_event:
+        # I0628 15:53:02.269804       1 operator/operator.go:133] <message>
```
Comment (juliayakovlev) [inline sdcm/sct_events/operator.py:42]: I suggest to add an explanation here, what is "I0628"

File: sdcm/sct_events/operator.py [hunk 4]
```diff
@@ -28,74 +28,49 @@ class ScyllaOperatorLogEvent(LogEvent):
...
-            if log_en_data is None:
-                continue
-            if attr_name == 'timestamp':
-                log_en_data = datetime.strptime(log_en_data, "%Y-%m-%dT%H:%M:%S.%fZ").timestamp()
-            event_data[attr_name] = log_en_data
+        self.source = source
+        self.line_number = None
+
+    # pylint: disable=too-many-locals
+    def add_info(self: T_log_event, node, line: str, line_number: int) -> T_log_event:
```
Comment (juliayakovlev) [inline sdcm/sct_events/operator.py:41]: I can't find where do you use `line_number` in this function

File: sdcm/sct_events/operator.py [hunk 5]
```diff
@@ -28,74 +28,49 @@ class ScyllaOperatorLogEvent(LogEvent):
...
+        # I0628 15:53:02.269804       1 operator/operator.go:133] <message>
+        splits = line.split(maxsplit=4)
+        if len(splits) != 5 or len(splits[0]) != 5:
+            return self
+        type_month_date, time_string, _, self.source, self.message = splits
```
Comment (juliayakovlev) [inline sdcm/sct_events/operator.py:46]: Do we need to analyze "I0628" and fetch the severity of the line (Info, Error, for example)?

**Review by juliayakovlev — COMMENTED** [review]
few comments

File: sdcm/sct_events/operator.py (hunk 5 above)
Comment (dkropachev) [reply sdcm/sct_events/operator.py:46]: Nope, severity should go from the pattern, not from the log message

File: sdcm/sct_events/operator.py (hunk 3 above)
Comment (dkropachev) [reply sdcm/sct_events/operator.py:42]: I believe that `type_month_date` is descriptive enough

File: sdcm/sct_events/operator.py (hunk 4 above)
Comment (dkropachev) [reply sdcm/sct_events/operator.py:41]: I don't see any reason to have it there, we use it only for backtrace and log event printing, which have nothing to do with this particular case.

Comment (juliayakovlev) [reply sdcm/sct_events/operator.py:41]: So,please, remove it

File: sdcm/sct_events/operator.py (hunk 3 above)
Comment (juliayakovlev) [reply sdcm/sct_events/operator.py:42]: Not sure that it's so descriptive especially for a person that new in kubernetes

File: sdcm/sct_events/operator.py (hunk 4 above)
Comment (dkropachev) [reply sdcm/sct_events/operator.py:41]: I can't it should match original definition.

File: sdcm/sct_events/operator.py (hunk 3 above)
Comment (dkropachev) [reply sdcm/sct_events/operator.py:42]: added

**Review by juliayakovlev — APPROVED** [review]

**Pull request merged**